pnet_datalink/src/lib.rs

```diff
@@ -216,6 +216,26 @@ impl NetworkInterface {
     pub fn is_loopback(&self) -> bool {
         self.flags & (pnet_sys::IFF_LOOPBACK as u32) != 0
     }
+    #[cfg(any(target_os = "macos", target_os = "linux"))]
+    pub fn is_up(&self) -> bool {
```

Comment:
Are these not valid for other unix distributions? Could we not just tag them with `#[cfg(unix)]`?

Reply:
i think that will be work on other bsd-like system too, 
but i'm not sure.

Reply:
I'd like these to not have any `#[cfg]` stuff. The methods are applicable on all platforms, and should be available on all platforms.

Perhaps these should return a tri-state, true/false/unknown instead?

Reply:
If they *always* return `unknown` on other platforms I don't see the point in having them on those platforms.

That being said, I think there is great value in keeping the library cross platform! So there should be some work done trying to make them work on all platforms.

If they turn out to not be able to implement on some platforms (non unix it looks like for now) then I think they should not be conditionally compiled like this. Instead I think it's better to move them to a unix specific trait that is implemented for this struct. Then the trait is conditionally compiled. The difference is the user must explicitly import something with `unix::` in the path to use it, which will make it very clear they are not in cross-platform land any longer. That is how the standard library usually deals with this. See these exemples:
* https://doc.rust-lang.org/stable/std/os/unix/fs/trait.FileExt.html
* https://doc.rust-lang.org/stable/std/os/unix/ffi/trait.OsStrExt.html
* And some other libraries: https://docs.rs/duct/0.9.2/duct/unix/trait.HandleExt.html (full disclosure, I'm the author of that trait in duct)

Reply:
They can be implemented at a later point - all these flags are applicable on all platforms.

If this were platform-specific functionality, then `#[cfg]` would be fine, but since it isn't, it should be available for everything, even if it's not implemented yet.

Reply:
Just afraid it will be hard to know if one can rely on them or not. Making all types `Option<bool>` becomes messy. Afraid they will be left unimplemented on some platforms forever and then we have complicated the type for no reason. Forcing Linux and macOS implementers to add extra error handling code even if it's 100% certain it will work on their platforms. Which is why I prefer the trait approach. They can still be moved to the general type as soon as they are available for all platforms, an implementer only have to remove the trait import.